src/index.js

```diff
@@ -7,36 +7,36 @@ const Hubot = require('hubot/es2015');
 process.setMaxListeners(0);
 
 class MockResponse extends Hubot.Response {
-  sendPrivate(/* ...strings*/) {
-    const strings = [].slice.call(arguments, 0);
-
-    this.robot.adapter.sendPrivate.apply(this.robot.adapter, [this.envelope].concat(strings));
+  sendPrivate(...strings) {
+    this.robot.adapter.sendPrivate(this.envelope, ...strings);
   }
 }
 
 class MockRobot extends Hubot.Robot {
-  constructor(httpd) {
-    if (httpd == null) { httpd = true; }
-    super(null, null, httpd, 'hubot');
+  constructor(httpd = false, botName = null, roomName = null) {
+    super('test', httpd ?? false, botName ?? 'hubot', null);
 
+    this.roomName = roomName ?? 'room1';
     this.messagesTo = {};
 
     this.Response = MockResponse;
   }
 
-  messageRoom(roomName, str) {
-    if (roomName == this.adapter.name) {
-      this.adapter.messages.push(['hubot', str]);
+  async messageRoom(room, ...strings) {
+    if (room === this.adapter.name) {
+      strings.forEach((str) => this.adapter.messages.push(['hubot', str]));
     } else {
-      if (!(roomName in this.messagesTo)) {
-        this.messagesTo[roomName] = [];
+      if (!(room in this.messagesTo)) {
+        this.messagesTo[room] = [];
       }
-      this.messagesTo[roomName].push(['hubot', str]);
+      strings.forEach((str) => this.messagesTo[room].push(['hubot', str]));
```

Comment:
I would think you'd need to use `this.name` not 'hubot' on these?

Reply:
You are right, i updated it. Good catch, thank you :)